Zend/zend_variables.c

```diff
@@ -37,18 +37,18 @@ static void ZEND_FASTCALL zend_empty_destroy(zend_reference *ref);
 typedef void (ZEND_FASTCALL *zend_rc_dtor_func_t)(zend_refcounted *p);
 
 static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
-	/* IS_UNDEF        */ (zend_rc_dtor_func_t)zend_empty_destroy,
-	/* IS_NULL         */ (zend_rc_dtor_func_t)zend_empty_destroy,
-	/* IS_FALSE        */ (zend_rc_dtor_func_t)zend_empty_destroy,
-	/* IS_TRUE         */ (zend_rc_dtor_func_t)zend_empty_destroy,
-	/* IS_LONG         */ (zend_rc_dtor_func_t)zend_empty_destroy,
-	/* IS_DOUBLE       */ (zend_rc_dtor_func_t)zend_empty_destroy,
-	/* IS_STRING       */ (zend_rc_dtor_func_t)zend_string_destroy,
-	/* IS_ARRAY        */ (zend_rc_dtor_func_t)zend_array_destroy,
-	/* IS_OBJECT       */ (zend_rc_dtor_func_t)zend_objects_store_del,
-	/* IS_RESOURCE     */ (zend_rc_dtor_func_t)zend_list_free,
-	/* IS_REFERENCE    */ (zend_rc_dtor_func_t)zend_reference_destroy,
-	/* IS_CONSTANT_AST */ (zend_rc_dtor_func_t)zend_ast_ref_destroy
+	[IS_UNDEF] =        (zend_rc_dtor_func_t)zend_empty_destroy,
```

Comment:
well, if we are to use this C99 feature, would assignments such as
`[IS_UNDEF...IS_DOUBLE] = ` work ?
otherwise, while the change itself is correct, as is I feel it does not real bring something to the table. What do you think ?

Reply:
Isn't that range thing a GNU extension?

Reply:
that s possible, maybe MSVC does not support it (would not surprise me).

Reply:
I ll let @Girgias decide if it s worth merging.

Reply:
> Isn't that range thing a GNU extension?

It is. https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html
"To initialize a range of elements to the same value, write ‘[first ... last] = value’. **This is a GNU extension.**"

It's tempting, but in this case it's not worth deviating from the standard.